src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -231,22 +239,38 @@ public MedicalInventory getInventoryByReference(String reference) throws OHServi
 	/**
 	 * Verify if the object is valid for CRUD and return a list of errors, if any.
 	 *
-	 * @param medInventory
+	 * @param medicalInventory the {@link MedicalInventory} to validate
+	 * @param checkReferenceExists if {@code true}, it verifies the reference uniqueness
 	 * @throws OHDataValidationException
 	 */
-	private void validateMedicalInventory(MedicalInventory medInventory) throws OHDataValidationException {
+	private void validateMedicalInventory(MedicalInventory medicalInventory, boolean checkReferenceExists) throws OHDataValidationException {
 		List<OHExceptionMessage> errors = new ArrayList<>();
 		LocalDateTime tomorrow = LocalDateTime.now().plusDays(1);
-		String reference = medInventory.getInventoryReference();
-		if (medInventory.getInventoryDate() == null) {
+		String reference = medicalInventory.getInventoryReference();
+		if (medicalInventory.getInventoryDate() == null) {
 			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.pleaseinsertavalidinventorydate.msg")));
 		}
-		if (medInventory.getInventoryDate() != null && medInventory.getInventoryDate().isAfter(tomorrow)) {
+		if (medicalInventory.getInventoryDate() != null && medicalInventory.getInventoryDate().isAfter(tomorrow)) {
 			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.notdateinfuture.msg")));
 		}
 		if (reference == null || reference.equals("")) {
 			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.mustenterareference.msg")));
 		}
+		/*
+		 * Despite MINVT_REFERENCE is VARCHAR(50), this field will be copied into MMV_REFNO and concatenate to either "-charge" or "-discharge" suffixes, hence
+		 * the limit to 40. For similarity, we keep the same validation method for "ward" type inventories, despite the field would be copied into MMVN_DESC
+		 * VARCHAR(100) in those cases.
+		 */
+		if (reference.length() > 40) {
```

Comment:
Line 256 checks `reference` for `null` so at this line it could be `null` thus throwing a NPE.  Shouldn't the be in a nested `else` clause?

Reply:
good one, fixed... thanks!